File: _정답코드/5주차/02_소수_구간합/Solution.java
class Solution {
    private static final int MAX = 5000000;
    private static boolean[] isPrime = new boolean[MAX + 1];
    
    public int solution(int[] numbers) {
        isPrimeInit();
        
        int n = numbers.length;
        int[] prefixSum = new int[n + 1];
        
        for (int i = 0; i < n; i++) {
            prefixSum[i + 1] = prefixSum[i] + numbers[i];
        }
        
        int count = 0;
        for (int i = 1; i <= n; i++) {
            for (int j = i; j <= n; j++) {
                int sum = prefixSum[j] - prefixSum[i - 1];
                if (isPrime[sum]) {
                    count++;
                }
            }
        }
        
        return count;
    }

    private void isPrimeInit() {
        for (int i = 2; i <= MAX; i++) {
            isPrime[i] = true;
        }
        
        for (int i = 2; i * i <= MAX; i++) {
            if (isPrime[i]) {
                for (int j = i * i; j <= MAX; j += i) {
                    isPrime[j] = false;
                }
            }
        }
    }
}

File: _정답코드/5주차/02_소수_구간합/solution.py
MAX = 1000000
isPrime = [False] * (MAX + 1)

for i in range(2, MAX + 1):
    isPrime[i] = True

for i in range(2, int(MAX**0.5) + 1):
    if isPrime[i]:
        for j in range(i * i, MAX + 1, i):
            isPrime[j] = False

def solution(numbers):
    n = len(numbers)
    prefixSum = [0] * (n + 1)
    
    for i in range(n):
        prefixSum[i + 1] = prefixSum[i] + numbers[i]
    
    count = 0
    for i in range(1, n + 1):
        for j in range(i, n + 1):
            sum_val = prefixSum[j] - prefixSum[i - 1]
            if sum_val >= 2 and sum_val <= MAX and isPrime[sum_val]:
                count += 1
    
    return count

